workflow/commons/src/main/java/com/datastax/oss/dsbulk/workflow/commons/schema/DefaultReadResultMapper.java

```diff
@@ -33,36 +31,33 @@
 import com.datastax.oss.dsbulk.mapping.CQLWord;
 import com.datastax.oss.dsbulk.mapping.Mapping;
 import edu.umd.cs.findbugs.annotations.NonNull;
-import java.net.InetSocketAddress;
 import java.net.URI;
 import java.util.Set;
-import java.util.function.Supplier;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
 
 public class DefaultReadResultMapper implements ReadResultMapper {
 
-  private static final Logger LOGGER = LoggerFactory.getLogger(DefaultReadResultMapper.class);
-  private static final URI UNKNOWN_ROW_RESOURCE = URI.create("cql://unknown");
-
   private final Mapping mapping;
   private final RecordMetadata recordMetadata;
+  private final URI resource;
+  private final boolean retainRecordSources;
 
-  public DefaultReadResultMapper(Mapping mapping, RecordMetadata recordMetadata) {
+  public DefaultReadResultMapper(
+      Mapping mapping, RecordMetadata recordMetadata, URI resource, boolean retainRecordSources) {
     this.mapping = mapping;
     this.recordMetadata = recordMetadata;
+    this.resource = resource;
+    this.retainRecordSources = retainRecordSources;
   }
 
   @NonNull
   @Override
   public Record map(@NonNull ReadResult result) {
-    Row row = result.getRow().orElseThrow(IllegalStateException::new);
-    Supplier<URI> resource =
-        () ->
-            getRowResource(row, result.getExecutionInfo().orElseThrow(IllegalStateException::new));
```

Comment:
Simplifying this whole nonsense. Creating a URI for a row is just not possible, and not useful, so we are now defaulting to a basic `cql://keyspace/table` URI.